cpp/src/glm/ridge.cuh

```diff
@@ -84,17 +84,16 @@ void ridgeSVD(const raft::handle_t &handle, math_t *A, int n_rows, int n_cols,
   int U_len = n_rows * n_cols;
   int V_len = n_cols * n_cols;
 
-  raft::allocate(U, U_len);
-  raft::allocate(V, V_len);
-  raft::allocate(S, n_cols);
+  rmm::device_uvector<math_t> S_vector(n_cols, stream);
+  rmm::device_uvector<math_t> V_vector(V_len, stream);
+  rmm::device_uvector<math_t> U_vector(U_len, stream);
+  S = S_vector.data();
```

Comment:
Is it necessary to have these locals here? Seems like it might be cleaner to just pass S_vector.data() etc. as parameters to the two functions below and not create additional locals that are bare pointers

Reply:
I removed these locals that are not necessary. I left some of them when they are used multiple times to make the code shorter.